lib/core/src/lib/search-text/search-text-input.component.stories.ts

```diff
@@ -0,0 +1,195 @@
+            control: 'inline-radio',
+            options: ['collapsed', 'expanded'],
+            description: 'Default state.',
+            defaultValue: 'collapsed',
+            table: {
+                type: { summary: 'string' },
+                defaultValue: { summary: 'collapsed' }
+            }
+        },
+        collapseOnBlur: {
+            control: 'boolean',
+            description: 'Collapse search bar on blur.',
+            defaultValue: true,
+            table: {
+                type: { summary: 'boolean' },
+                defaultValue: { summary: 'true' }
+            }
+        },
+        showClearButton: {
+            control: 'boolean',
+            description: 'Toggles whether to show a clear button that closes the search.',
+            defaultValue: false,
+            table: {
+                type: { summary: 'boolean' },
+                defaultValue: { summary: 'false' }
+            }
+        },
+        placeholder: {
+            control: 'text',
+            description: 'Placeholder text to show in the input field.',
+            defaultValue: '',
+            table: {
+                category: 'HTML input attributes',
+                type: { summary: 'string' },
+                defaultValue: { summary: '' }
+            }
+        },
+        hintLabel: {
+            control: 'text',
+            description: 'Hint label.',
+            defaultValue: '',
+            table: {
+                category: 'HTML input attributes',
+                type: { summary: 'string' },
+                defaultValue: { summary: '' }
+            }
+        },
+        searchChange: {
+            action: 'searchChange',
+            description: 'Emitted when the search term is changed. The search term is provided in the "value" property of the returned object. If the term is less than three characters in length then it is truncated to an empty string.',
+            table: { category: 'Actions' }
+        },
+        submit: {
+            action: 'submit',
+            description: 'Emitted when the search is submitted by pressing the ENTER key.',
+            table: { category: 'Actions' }
+        },
+        selectResult: {
+            action: 'selectResult',
+            description: 'Emitted when the result list is selected.',
+            table: { category: 'Actions' }
+        },
+        reset: {
+            action: 'reset',
+            description: 'Emitted when the result list is reset.',
+            table: { category: 'Actions' }
+        },
+        searchVisibility: {
+            action: 'searchVisibility',
+            description: 'Emitted when the search visibility changes. True when the search is active, false when it is inactive.',
+            table: { category: 'Actions' }
+        }
+    }
+} as Meta;
+
+const template: Story<SearchTextInputComponent> = (args: SearchTextInputComponent) => ({
+    props: args
+});
+
+export const searchTextInput = template.bind({});
```

Comment:
just out of curiosity, is this needed? (I saw that also in different stories) Generally speaking we cannot rebind `this` in arrow function (arrow functions do not have its own `this`, it points to [lexical this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this))

If we would like to change the `this` I think we can have something like:

```
template = function(args) {...}
```

Reply:
This is the Storybook's way of defining stories and inferring their names - we create a template and export functions binded to it.  [more here](https://storybook.js.org/docs/angular/writing-stories/introduction#using-args)

Reply:
👌

`Template.bind({}) is a standard JavaScript technique for making a copy of a function. We copy the Template so each exported story can set its own properties on it.
`